utils/bimap/bimap.go

```diff
@@ -0,0 +1,169 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package bimap
+
+import (
+	"sync"
+
+	"github.com/ava-labs/avalanchego/utils"
+)
+
+var _ BiMap[int, int] = (*biMap[int, int])(nil)
+
+type Entry[K, V any] struct {
+	Key   K
+	Value V
```

Comment:
If a key/value has a usable zero value it's not clear what was evicted or not, e.g if it's an `int`.... feels like we might need a bool flag to indicate if it's evicted or not but this sounds gross

Reply:
Talked offline - we don't use the zero value as a flag, so we don't need any special flags.